src/main/scala/where/Predicate.scala

```diff
@@ -1,10 +1,34 @@
 package com.originate.scalypher.where
 
 import com.originate.scalypher.ConstantString
+import com.originate.scalypher.path.AnyNode
+import com.originate.scalypher.path.AnyRelationship
+import com.originate.scalypher.path.Path
 import com.originate.scalypher.ToQuery
 
-sealed trait Predicate extends ToQuery
+case class PredicateWithProjection(
+  predicate: Predicate,
+  projection: PathProjection
+) {
+  def where(f: ObjectReference => Where): PredicateCondition =
+    PredicateCondition(predicate, projection, f)
+}
+
+sealed trait Predicate extends ToQuery {
+  def nodesIn(path: Path): PredicateWithProjection =
+    PredicateWithProjection(this, Nodes(path))
+
+  def nodeIn(path: Path): PredicateWithProjection =
+    PredicateWithProjection(this, Nodes(path))
+
+  def relationshipsIn(path: Path): PredicateWithProjection =
+    PredicateWithProjection(this, Relationships(path))
+
+  def relationshipIn(path: Path): PredicateWithProjection =
+    PredicateWithProjection(this, Relationships(path))
+}
+
 case object All extends ConstantString("ALL") with Predicate
 case object Any extends ConstantString("ANY") with Predicate
-case object NonePredicate extends ConstantString("NONE") with Predicate
+case object No extends ConstantString("NONE") with Predicate
```

Comment:
having trouble figuring out the best name for this here :/
```scala
No nodesIn path where { node => ... }
NotAny nodesIn path where { node => ... }
```
obviously can't collide with `None` :'(